functions/src/index.ts

```diff
@@ -129,120 +151,105 @@ exports.calculateAqi = functions.pubsub
+        readingDocId: sensorDoc.id,
+        lastValidAqiTime: sensorDocData.lastValidAqiTime ?? null,
+        lastSensorReadingTime: sensorDocData.lastSensorReadingTime ?? null,
+        isActive: sensorDocData.isActive ?? true,
+        nowCastPm25: NaN,
+        aqi: NaN,
+        isValid: false,
+      };
 
-      // Get current sensor readings
-      const hourlyAverages = await getHourlyAverages(docId);
-      const cleanedAverages = cleanAverages(hourlyAverages);
+      if (currentSensorData.isActive) {
+        // Get current sensor readings
+        const hourlyAverages = await getHourlyAverages(sensorDoc.id);
+        const cleanedAverages = cleanAverages(hourlyAverages);
 
-      // NowCast formula from the EPA requires 2 out of the last 3 hours
-      // to be available
-      let validEntriesLastThreeHours = 0;
-      const THREE_HOURS = 3;
-      for (
-        let i = 0;
-        i < Math.min(THREE_HOURS, cleanedAverages.readings.length);
-        i++
-      ) {
-        if (!Number.isNaN(cleanedAverages.readings[i])) {
-          validEntriesLastThreeHours++;
+        // NowCast formula from the EPA requires 2 out of the last 3 hours
+        // to be available
+        let validEntriesLastThreeHours = 0;
+        const THREE_HOURS = 3;
+        for (
+          let i = 0;
+          i < Math.min(THREE_HOURS, cleanedAverages.readings.length);
+          i++
+        ) {
+          if (!Number.isNaN(cleanedAverages.readings[i])) {
+            validEntriesLastThreeHours++;
+          }
         }
-      }
-      const NOWCAST_RECENT_DATA_THRESHOLD = 2;
-      const containsEnoughInfo =
-        validEntriesLastThreeHours >= NOWCAST_RECENT_DATA_THRESHOLD;
+        const NOWCAST_RECENT_DATA_THRESHOLD = 2;
+        const containsEnoughInfo =
+          validEntriesLastThreeHours >= NOWCAST_RECENT_DATA_THRESHOLD;
 
-      // If there's enough info, the sensor's data is updated
-      // If there isn't, we send the default AQI buffer element
-      let aqiBufferData = defaultAqiBufferElement; // New data to add
+        // If there's enough info, the sensor's data is updated
+        // If there isn't, we send the default AQI buffer element
+        const aqiBufferData = defaultAqiBufferElement; // New data to add
 
-      // Defaults for a sensor
-      let latitude = NaN;
-      let longitude = NaN;
-      let nowCastPm25 = NaN;
-      let aqi = NaN;
-      let isValid = false;
-      let lastValidAqiTime: FirebaseFirestore.Timestamp | null = null;
+        // If there is not enough info, the sensor's status is not valid
+        if (containsEnoughInfo) {
+          // Only calculate the NowCast PM 2.5 value and the AQI if there is enough data
+          const nowCastPm25Result = NowCastConcentration.fromCleanedAverages(
+            cleanedAverages
+          );
+          currentSensorData.aqi = aqiFromPm25(nowCastPm25Result.reading);
+          currentSensorData.latitude = nowCastPm25Result.latitude;
```

Comment:
I see that we update the latitude, longitude, valid, etc in the currentReadings doc, but we don't do it in the sensor doc. Is that a problem?

Reply:
Ex: if a sensor moves, will it ever be updated in the sensor docs or only in current readings?